dymos/transcriptions/transcription_base.py

```diff
@@ -399,17 +399,53 @@ def setup_solvers(self, phase):
         """
         raise NotImplementedError(f'Transcription {self.__class__.__name__} does not implement method setup_solvers.')
 
-    def configure_solvers(self, phase):
+    def configure_solvers(self, phase, requires_solvers=None):
         """
-        Configure the solvers for this transcription.
+        Configure the solvers.
 
         Parameters
         ----------
         phase : dymos.Phase
             The phase object to which this transcription instance applies.
-        """
-        raise NotImplementedError(f'Transcription {self.__class__.__name__} does not implement method '
-                                  f'configure_solvers.')
+        requires_solvers : dict[str: bool]
+            A dictionary mapping a string descriptor of a reason why a solver is required,
+            and whether a solver is required.
+        """
+        if not phase.options['auto_solvers']:
+            return
+
+        req_solvers = {'implicit duration': self._implicit_duration}
+
+        if requires_solvers is not None:
+            req_solvers.update(requires_solvers)
+
+        reasons_txt = '\n'.join(f'    {key}: {val}' for key, val in req_solvers.items())
```

Comment:
Intersting that the user can put whatever they want in this sentence. Looks like it also prints out for every phase in the model?

Reply:
The intention was to try to include a "reason" in the warning that gets printed out, so allowing the string in here seemed like the quickest way to do it.

And yes while a "user" could do this, in this context "user" really means "transcription developer".

Reply:
This would print out for every phase where this behavior is applicable.  I agree that that's somewhat noisy, but users are capable of squelching it by explicitly setting the solver themselves.